### PR TITLE
cc2538: fix xtimer

### LINKED ISSUE #6419: cpu/cc2538: timer broken?
On remote boards it appears that `xtimer_usleep` hangs. Didn't test on other cc2538 boards. Even *tests/periph_timer* acts weird as it  can't initialise all timers:

```
2017-01-19 10:19:11,538 - INFO # Available timers: 4
2017-01-19 10:19:11,538 - INFO # 
2017-01-19 10:19:11,539 - INFO # Testing TIMER_0:
2017-01-19 10:19:11,540 - INFO # TIMER_0: initialization successful
2017-01-19 10:19:11,540 - INFO # TIMER_0: stopped
2017-01-19 10:19:11,541 - INFO # TIMER_0: set channel 0 to 5000
2017-01-19 10:19:11,542 - INFO # TIMER_0: set channel 1 to 10000
2017-01-19 10:19:11,542 - INFO # TIMER_0: starting
2017-01-19 10:19:11,621 - INFO # TIMER_0: channel 0 fired at SW count    12309 - init:    12309
2017-01-19 10:19:11,623 - INFO # TIMER_0: channel 1 fired at SW count    76926 - diff:    64617
2017-01-19 10:19:11,624 - INFO # 
2017-01-19 10:19:11,624 - INFO # Testing TIMER_1:
2017-01-19 10:19:11,625 - INFO # TIMER_1: ERROR on initialization - skipping
2017-01-19 10:19:11,626 - INFO # 
2017-01-19 10:19:11,626 - INFO # 
2017-01-19 10:19:11,627 - INFO # Testing TIMER_2:
2017-01-19 10:19:11,628 - INFO # TIMER_2: ERROR on initialization - skipping
2017-01-19 10:19:11,628 - INFO # 
2017-01-19 10:19:11,628 - INFO # 
2017-01-19 10:19:11,629 - INFO # Testing TIMER_3:
2017-01-19 10:19:11,629 - INFO # TIMER_3: ERROR on initialization - skipping
2017-01-19 10:19:11,630 - INFO # 
2017-01-19 10:19:11,630 - INFO # 
2017-01-19 10:19:11,630 - INFO # TEST FAILED
```

### PR DESCRIPTION
**Pull request by smlng**
This PR fixes #6419, at least for me. 

It basically reverts xtimer configuration introduces in #5608, and resets XTIMER_SHIFT to 0 and XTIMER_HZ 1000000 (1 MHz).  I have to admit, I'm not sure why this helps, but it does. In the last days I made extensive tests using the hardware timer and xtimer.

The cc2538 has 4 timers each can be configured in 16Bit mode with 2 channels and 32 Bit mode with 1 channel. For the latter only a frequency of 16 MHz is allowed, as prescaler is only available in 16 Bit mode. However, using 16Bit timer with 16Mhz or a 32Bit timer with 16 MHz the tests `xtimer_drift` and `xtimer_periodic_wakeup` always fail, and `xtimer_usleep` for 16Bit + 16MHz, too.  All tests succeeded when using a 16 Bit timer with 1 MHz (which is the old setting).

Can someone please confirm?

### CODE REVIEW

**Comment by kYc0o** [comment]
I'm trying to test in a cc2538dk board applying the same modifications without success... Why it should only work for remote boards?

**Comment by smlng** [comment]
mhm, I'm unsure as well - because I was pretty sure I tested various settings yesterday, including 16Bit with 1MHz and it didn't work. However, today it worked 😕

**Comment by smlng** [comment]
I think we have cc2538dk here, too - so I'll try to test that, too

**Comment by PeterKietzmann** [comment]
`xtimer_hang` or `xtimer_usleep` still not running on remote-revb

**Comment by A-Paul** [comment]
I confirm that both tests are also not running on my *cc2538dk*.

**Comment by kYc0o** [comment]
I just tested all xtimer tests and no one worked. I have a remote-reva.

**Comment by smlng** [comment]
oops, during my debugging and tests session I found that only the cc2538 was missing the `periph_pm` feature (other cortex cpus have that) so I added a Makefile.feature. I forgot about that, so it lay around in my tree but wasn't commited. It seems that this is the missing piece!

Please test again @PeterKietzmann, @kYc0o, and @A-Paul !

**Comment by smlng** [comment]
I think this was caused when the old `lmp` code was removed in #6160 - but the cc2538 was forgotten then

**Comment by smlng** [comment]
to give a bit more rational on why `periph_pm` is needed: during debugging I found (with lots of debug output enabled) that everything seems to finally hang inside the idle thread. So I looked into that and since `lpm` removed, I guess it causes a deep sleep mode or something - in any case though the hardware timer fires initially it fails to fire again after idle thread.

File: cpu/cc2538/Makefile.features [hunk 1]
```diff
@@ -0,0 +1 @@
+FEATURES_PROVIDED += periph_pm
```
Comment (haukepetersen) [inline cpu/cc2538/Makefile.features]: unrelated?!

Comment (smlng) [inline cpu/cc2538/Makefile.features]: NO its *one* _essential_ part of the fix! Read my explanation above, otherwise everything gets stuck in idle thread. I could make a separate PR for this, but its already a commit and as said its part of the fix.

**Comment by PeterKietzmann** [comment]
I ran `xtimer_hang`, `xtimer_usleep` and `xtimer_drift` with success. The last one ran for about half an hour with no significant drift. @A-Paul please test the xtimer on your 2538dk again. @haukepetersen do you have objections against the PM stuff?

**Comment by A-Paul** [comment]
On *cc2538dk* all *tests/xtimer_** look good now.

**Comment by PeterKietzmann** [comment]
Jenkins fails unrelated and both Murdocks succeeded. ACK and go

**Comment by smlng** [comment]
finally 🎉

**Comment by kYc0o** [comment]
Thanks @smlng !

File: boards/openmote-cc2538/include/board.h [hunk 2]
```diff
@@ -67,8 +67,8 @@
 #define XTIMER_DEV          (0)
 #define XTIMER_CHAN         (0)
 #define XTIMER_WIDTH        (16)
-#define XTIMER_SHIFT        (4)
-#define XTIMER_HZ           (16000000UL)
+#define XTIMER_SHIFT        (0)
+#define XTIMER_HZ           (10000000UL)
```
Comment (smlng) [inline boards/openmote-cc2538/include/board.h]: this was already fixed before merge! Reload the page and have another look?
